### ISSUE TITLE
How to calculate brier score for given set of probabilities?

### ISSUE THREAD

**Issue by sahasud1** [tagteam/riskRegression#28]
Hi Thomas,

Thank you so much for a great package. I would like to calculate brier score for a given set of survival probabilities. Say, I have a survival dataset where ```time``` and ```status``` variable are given along with some survival probabilities at (say) ```time = 12```, calculated based on some indices for each individual in the dataset. Now, how I can calculate brier score for each set of indices given the data. Here is the code

    set.seed(99)
    library(dplyr)
    library(survival)
    library(prodlim) 

    # simulated survival data data
    data <- SimSurv(100)

    # add two incides
    data$index1<-runif(100, 0.4, 0.9)
    data$index2<-runif(100, 0.5, 1)

Basically, instead of some `coxph` based prediction, I already have prediction probabilities provided like in `index1` and `index2`.  How can I calculate the brier score for each of the index. 

Thanks again,
Sudip

**Comment by tagteam** [comment]
based on your example, lets assume that index1 and index2 are predicted
risks at times 1 and 2 respectively. the following works:

set.seed(99)
library(dplyr)
library(survival)
library(prodlim)
library(riskRegression) 

# simulated survival data data
data <- SimSurv(100)

# add two incides
index1<-runif(100, 0.4, 0.9)
index2<-runif(100, 0.5, 1)
x =
Score(list(cbind(index1,index2)),data=data,formula=Surv(time,event)~1,times=c(1,2))

does that answer your question?

**Comment by sahasud1** [comment]
That perfectly answers my question. Thanks a lot!

Sudip

**Comment by svd09** [comment]
Hi Thomas,

Thank you for your package and book (with Kattan). Like the above example, I have predicted probabilities and an external dataset. I would like to externally validate the predicted probabilities from that model on my dataset and calculate AUC, plot the ROC and calibration curves for that model. I have provided example code using the lung dataset in the survival package.

My issue is I cannot understand how to put the predicted probabilities into the Score function to get the results that I need.
Thank you - 

example here - 

# Prepare script for example on validation using the riskRegression package.
# use lung data set from the survival package

library(easypackages)
libraries(c("tidyverse","tidylog","riskRegression",
"ggthemes","ggsci","survival"))

library(survival)


# now to get the data set and then split into the test and train data.

df = lung

df -> lung

df$id = 1:nrow(df)

glimpse(df)

df1 = df[1:100, ]

df2 = df[101:max(df$id), ]

glimpse(df1); glimpse(df2)

# so now we have two data df1 and df2.
# am going to fit a model to df1 and then use that model to obtain
# predictions for df2. 
# then use these predictions to compare it to the null model 
# and hence see how good they work for predicting outcome in df2. 

# fit a cox model for df1.

cox_model = coxph(Surv(time, status) ~ age + sex + ph.ecog  
,  data = df1, x = T, y = T)

# now to get the predicted probability of mortality

df2$pred_surv = predict(object = cox_model, 
newdata = df2, 
times = 240, type = "survival")

# now these are survival probabilites, 
# for the regular time to event models, 
# predicted mortality = 1 - predicted survival 

# so to get the predicted mortality from the survival probabilities 

df2$pred_mort = 1 - df2$pred_surv

# now we can get the same predicted probabilites for the null model.

cox_null = coxph(Surv(time, status ) ~ 1, data = df2,
x = T, y = T)

cox_null

df2$pred_surv_null = predict(cox_null,
times = 240, type = "survival")

df2$pred_surv_null

df2$pred_mort_null = 1 - df2$pred_surv_null

# make sure that these are different 

par(mfrow = c(1,2))

boxplot(x = df2$pred_mort_null)
boxplot(x =df2$pred_mort)

summary(df2$pred_mort)
summary(df2$pred_mort_null)

# you can see that there is some change in the fitted model compared 
# to the null model. 

# now the first step would be see the distribution of predicted mortality risk 

par(mfrow = c(2,1))

hist(df2$pred_mort_null, col = "red",
main = "Predicted mortality at 240 days in the fitted Cox model")
hist(df2$pred_mort, col = "blue",
main = "Predicted mortality at 240 days in the baseline null Cox model")

# now to plot the predicted probabilities for each variable included in the model 
# the var included in the models are - sex, age, ph.ecog


par(mfrow = c(2,2))

boxplot(df2$pred_mort ~ df2$sex)
plot(x = df2$age, y = df2$pred_mort)
boxplot( df2$pred_mort ~ df2$ph.ecog)

# now we need to calculate the Brier score as the measure of validation 
# we can use the predicted probabilities obtained to calculate the Brier score 
# if we do not have access to the original data and cannot fit 2 different models.

# so we can put both model predicted probabilites into the object as a list and then 
# obtain the Brier score at that particular time point

null = df2$pred_mort_null
test = df2$pred_mort


brier_score =   riskRegression::Score(list(test, null),
                        data=df2,
                        formula=Surv(time,status) ~ 1,
                        times=c(240, 360), metrics = "brier")

brier_score


roc = riskRegression::Score(
    object = list(test),
    data = df2, 
    formula = Surv(time, status) ~ 1, 
    times = 240, 
    plots = "ROC"
)  # here I continue to get an error message. Thank you again for your excellent book and package
